src/ops.jl

```diff
@@ -0,0 +1,78 @@
+# Default implementations of ONNX operations
+
+import NNlib
+import Flux
+
+
+conv(x, w; kw...) = NNlib.conv(x, w; kw...)
+
+function conv(x, w, b; kw...)
+    bias_size = (ntuple(_ -> 1, length(kw.stride))..., :, 1)
+    b = reshape(b, bias_size)
+    return conv(x, w; kw...) .+ b
+end
+
+
+function onnx_gemm(A, B, C; tA=0, tB=false, α=1, β=1)
+    A = Bool(tA) ? A' : A
+    B = Bool(tB) ? B' : B
+    # note: order of arguments reversed due to row-major layout
+    return α * B * A .+ β * C
+end
+
+function onnx_gemm(A, B; tA=0, tB=0, α=1)
+    A = Bool(tA) ? A' : A
+    B = Bool(tB) ? B' : B
+    # note: order of arguments reversed due to row-major layout
+    return α * B * A
+end
+
+# Julia-friendly flatten
+function flatten(x; dim=ndims(x) - 1)
+    sz = size(x)
+    keep = dim < ndims(x) ? sz[dim + 1:end] : 1
+    return reshape(x, :, keep...)
+end
+
+# ONNX-specific flatten
+function onnx_flatten(x; axis=1)
+    dim = axis >= 0 ? ndims(x) - axis + 1 : axis + 1
+    return flatten(x; dim=dim)
+
+end
+
+add(xs...) = broadcast(+, xs...)
```

Comment:
`+` itself may be just as good if not better of an implementation for `add`, as it doesn't require broadcasting (which can be slow for GPU compilation when a lot of operands are involved, and may be slower for AD as well).

Reply:
A bit tangential, but is this the "official" way for elementwise addition today? 

I have somehow gotten the impression that automatic broadcasting of `+` was some kind of leftover from earlier days which should preferably not be used anymore. I suppose the practicalities of GPU compilation and AD are what they are though...

Reply:
No, I was just wrong. I assumed GPUArrays had a custom kernel for array `+`, but it falls back to the AbstractArray broadcasting path. ChainRules does have a [definition](https://github.com/JuliaDiff/ChainRules.jl/blob/2e6491c9dd20608b725b7ecdca2dbe872cf7833b/src/rulesets/Base/arraymath.jl#L322) for multi-arg `+` and Zygote one for [2-arg](https://github.com/FluxML/Zygote.jl/blob/master/src/lib/array.jl#L745), but I have no idea how they perform relative to the equivalent broadcast rules.